### ISSUE TITLE
use AnyEvent::HTTP to stream watch etc

### ISSUE THREAD

**Issue by hexfusion** [hexfusion/perl-etcd3#28]
here is a demo to emulate the watch example from etcd docs.

```
#!/usr/bin/env perl

use utf8;

use strict;
use warnings;

use JSON;
use AnyEvent;
use AnyEvent::HTTP;
use Data::Dumper;

my $cb = AnyEvent->condvar;
my $uri = 'http://127.0.0.1:2379/v3alpha/watch';
my $body = '{"create_request": {"key":"Zm9v"} }';

http_request('POST', $uri,
    headers => {
        Accept => '*/*',
        'Content-Type' => 'application/json'
    },
    body => $body,
    on_header => sub {
        my($headers) = @_;
        print Dumper($headers);
    },
    on_body   => sub {
        my ($data, $hdr) = @_;
        print $data;
        1
    },
    sub {
        my (undef, $hdr) = @_;
        my $status = $hdr->{Status};
        if ($status == 200 || $status == 206 || $status == 416) {
            # download ok || resume ok || file already fully downloaded
            $cb->(1, $hdr);
        } else {
            $cb->(undef, $hdr);
        }
    }
);

$cb->recv;
```

This ability will be critical to auth etc.  Probably going to need to adjust most or all requests away from HTTP::Tiny to this.

**Comment by hexfusion** [comment]
This will resolve https://github.com/hexfusion/perl-etcd3/issues/7

**Comment by hexfusion** [comment]
closed via 6a48f12218c9 + friends
